atom/browser/api/atom_api_web_contents.cc

```diff
@@ -919,10 +923,6 @@ bool WebContents::OnMessageReceived(const IPC::Message& message) {
 // be destroyed on close, and WebContentsDestroyed would be called for it, so
 // we need to make sure the api::WebContents is also deleted.
 void WebContents::WebContentsDestroyed() {
-  // This event is only for internal use, which is emitted when WebContents is
-  // being destroyed.
-  Emit("will-destroy");
```

Comment:
This changes the meaning of `will-destroy`, for webview it is emitted before destroying `WebContents`, however for normal window now it is emitted after `WebContents` gets destroyed. While this works, it makes the closing logic more difficult to understand.

How about solving the crash with a fix like https://github.com/electron/brightray/pull/284?